specutils/spectra/spectrum_collection.py

```diff
@@ -0,0 +1,90 @@
+import logging
+from collections import MutableSequence
+
+import astropy.units as u
+import numpy as np
+from astropy.nddata import NDIOMixin, NDUncertainty
+
+from .spectrum1d import Spectrum1D
+
+__all__ = ['SpectrumCollection']
+
+
+class SpectrumCollection(MutableSequence, NDIOMixin):
+    """
+    A container class for :class:`~specutils.Spectrum1D` objects. This allows 
+    for operations to be performed over a set of spectrum objects. This class 
+    behaves as if it were a :class:`~specutils.Spectrum1D` and supports
+    pythonic collection operations like slicing, deleting, and inserting.
+
+    Note
+    ----
+    Items in this collection must currently be the same shape. Items are not
+    automatically resampled onto a shared grid. Currently, users should resample
+    their spectra prior to creating a :class:`~specutils.SpectrumCollection`
+    object.
+
+    Parameters
+    ----------
+    items : list, ndarray
+        A list of :class:`~specutils.Spectrum1D` objects to be held in the
+        collection
+    """
+    def __init__(self, items):
+        # Enforce that the shape of each item must be the same
+        if not all((x.shape == items[0].shape for x in items)):
+            raise Exception("Shape of all elements must be the same.")
+
+        self._items = items
+
+    def __setitem__(self, index, value):
+        self._items[index] = value
+
+    def __delitem__(self, index):
+        del self._items[index]
+
+    def insert(self, index, value):
+        self._items.insert(index, value)
+
+    def __getitem__(self, index):
+        return self._items[index]
+
+    def __len__(self):
+        return len(self._items)
+
+    def _get_property(self, name):
+        val = [getattr(x, name) for x in self._items]
```

Comment:
While this is a clever quick solution to this problem, it has the disadvantage that it defeats the whole point of doing this: by doing a for-loop over all of the stored spectra, you don't get any numpy speed advantage.

That said, at *some* point you'd need to do a for-loop over input `Spectrum1D` objects.  But the user should be able to input their `flux` array *as an array*, in which case this technique won't work.

I realize this would require a lot of changes at a late hour.  If that doesn't seem feasible for 0.4 this current solution might work as sort of a starting point.  But it *must* be updated in 0.5 to realize the purpose of having a `SpectrumCollection` at all.

Reply:
The result of this _is_ an ndarray (except in cases where it's a `wcs` object) -- thus all the speed advantages of numpy. The maintaining of the list of spectra is done using python lists because that allows for easy slicing/deleting/appending/removing as opposed to rigid numpy data structures.